packages/components/src/components/post-icon/post-icon.tsx

```diff
@@ -104,71 +98,60 @@ export class PostIcon {
     checkEmptyOrType(this, 'scale', 'number');
   }
 
-  // Construct the icon url from different possible sources
-  private getUrl() {
-    let url: string;
-    const file = `${this.name}.svg`;
-
-    // the first definition object which defines a domain, will be used to set the domain of the file url
-    // the first definition object which defines a slug, will be used to set the slug of the file url
-    const urlDefinitions = [this.getUrlDefinition(this.base, 'both')];
-
-    if (IS_BROWSER) {
-      urlDefinitions.push(
-        this.getUrlDefinition(
-          document.head
-            .querySelector('meta[name="design-system-settings"][data-post-icon-base]')
-            ?.getAttribute('data-post-icon-base'),
-          'relative',
-        ),
-      );
-      urlDefinitions.push(
-        this.getUrlDefinition(document.querySelector('base[href]')?.getAttribute('href'), 'both'),
-      );
-    }
+  // Construct the icon URL according to the following rules:
+  // - URL = current domain + base[href] + meta[data-post-icon-base] (or component base)
+  // - If base[href] is not relative, it's used instead of current domain + base[href]
+  // - If meta[data-post-icon-base] (or component base) is not relative, it's used as is
+  private getUrl(): string {
+    const fileName = `${this.name}.svg`;
 
-    // in case no definition defines a domain, a relative url is used to load the icon
-    const origin = urlDefinitions.find(d => d.definesDomain)?.url?.origin;
-    // in case no definition defines a slug either, the cdn url is used as a fallback
-    const slug = urlDefinitions.find(d => d.definesSlug)?.url?.pathname;
+    if (!IS_BROWSER && !this.base) {
+      return `${CDN_URL}${fileName}`;
+    }
 
-    if (origin && slug) {
-      url = `${origin}${slug}${file}`;
-    } else if (!origin && slug) {
-      url = `${slug}${file}`;
-    } else {
-      url = `${CDN_URL}${file}`;
+    const currentDomain = IS_BROWSER ? window.location.origin : '';
+
+    const baseHref = IS_BROWSER
+      ? document.querySelector('base[href]')?.getAttribute('href') || ''
+      : '';
+    const isBaseHrefAbsolute = /^https?:\/\//.test(baseHref);
+
+    const iconBase =
+      this.base ||
+      (IS_BROWSER
+        ? document
+          .querySelector('meta[name="design-system-settings"]')
+          ?.getAttribute('data-post-icon-base') || ''
+        : '');
+    const isIconBaseAbsolute = /^https?:\/\//.test(iconBase);
+
+    let normalizedBaseHref = '';
+    if (baseHref) {
+      normalizedBaseHref = baseHref.endsWith('/') ? baseHref : `${baseHref}/`;
     }
 
-    return url;
-  }
+    let normalizedIconBase = '';
+    if (iconBase) {
+      normalizedIconBase = iconBase.endsWith('/') ? iconBase : `${iconBase}/`;
+    }
```

Comment:
This could be simplified with a sub-function:

```ts
const normalizedBaseHref = normalizeUrl(baseHref);
const normalizedIconBase = normalizeUrl(iconBase);

function normalizeUrl(url: string) {
  if (!url) return '';
  return url.endsWith('/') ? url : `${url}/`; 
} 
```